File: app/src/main/java/com/zihuan/app/xrv/RecycleLayout.java
package com.zihuan.app.xrv;

import android.content.Context;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import com.jcodecraeer.xrecyclerview.ProgressStyle;
import com.jcodecraeer.xrecyclerview.XRecyclerView;
import com.zihuan.app.R;


public class RecycleLayout {
    public static Context mContext;
    static RecycleLayout viewLayout;

    public static RecycleLayout builder(Context context) {
        if (viewLayout == null) viewLayout = new RecycleLayout();
        mContext = context;
        return viewLayout;
    }

    //    初始化xrv的配置
    public void initXrecyclerView(XRecyclerView xRecyclerView) {
        xRecyclerView.setLayoutManager(RecycleViewBuilder.initRecycleLayout().getLinLayoutManager(LinearLayoutManager.VERTICAL));
        initXrvConfig(xRecyclerView);
    }


    //    横向布局
    public void initHorizontal(RecyclerView recyclerView) {
        recyclerView.setLayoutManager(RecycleViewBuilder.initRecycleLayout().getLinLayoutManager(LinearLayoutManager.HORIZONTAL));
    }

    //    竖向布局
    public void initVertical(RecyclerView recyclerView) {
        recyclerView.setLayoutManager(RecycleViewBuilder.initRecycleLayout().getLinLayoutManager(LinearLayoutManager.VERTICAL));
    }

    //     可刷新的配置
    public void initXGrid(XRecyclerView xRecyclerView, int count) {
        xRecyclerView.setLayoutManager(RecycleViewBuilder.initRecycleLayout().getGridLayoutManager(count));
        initXrvConfig(xRecyclerView);
    }

    //    九宫格布局
    public void initGrid(RecyclerView recyclerView, int count) {
        recyclerView.setLayoutManager(RecycleViewBuilder.initRecycleLayout().getGridLayoutManager(count));
    }

    // xrv配置
    private void initXrvConfig(XRecyclerView xRecyclerView) {
        xRecyclerView.setRefreshProgressStyle(ProgressStyle.BallSpinFadeLoader);
        xRecyclerView.setLoadingMoreProgressStyle(ProgressStyle.Pacman);
        xRecyclerView.setArrowImageView(R.mipmap.iconfont_downgrey);
        xRecyclerView.setLoadingMoreEnabled(false);
    }

    private static class RecycleViewBuilder {
        static RecycleViewBuilder viewBuilder;

        public static RecycleViewBuilder initRecycleLayout() {
            if (viewBuilder == null) viewBuilder = new RecycleViewBuilder();
            return viewBuilder;
        }

        public LinearLayoutManager getLinLayoutManager(int orientation) {
            LinearLayoutManager layoutManager = new LinearLayoutManager(mContext);
            layoutManager.setOrientation(orientation);
            return layoutManager;
        }

        //九宫格布局
        public GridLayoutManager getGridLayoutManager(int count) {
            GridLayoutManager layoutManager = new GridLayoutManager(mContext, count);
            return layoutManager;
        }
    }
}

File: BaseWork/app/src/main/java/com/zihuan/app/base/RecycleLayout.kt
package com.zihuan.app.base

import android.annotation.SuppressLint
import android.content.Context
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.zihuan.baseadapter.RecyclerViewHolder

@SuppressLint("StaticFieldLeak")
object RecycleLayout {
   private var mContext: Context? = null

    //横向布局
    fun androidx.recyclerview.widget.RecyclerView.initHorizontal() {
        mContext = this.context
        this.layoutManager = RecycleLayout.RecycleViewBuilder.getLinLayoutManager(androidx.recyclerview.widget.LinearLayoutManager.HORIZONTAL, this)
    }

    fun androidx.recyclerview.widget.RecyclerView.initHorizontal(adapter: androidx.recyclerview.widget.RecyclerView.Adapter<RecyclerViewHolder>) {
        initHorizontal()
        this.adapter = adapter
    }


    //竖向布局
    fun androidx.recyclerview.widget.RecyclerView.initVertical(): androidx.recyclerview.widget.LinearLayoutManager {
        mContext = this.context
        this.layoutManager = RecycleLayout.RecycleViewBuilder.getLinLayoutManager(androidx.recyclerview.widget.LinearLayoutManager.VERTICAL, this)
        return this.layoutManager as androidx.recyclerview.widget.LinearLayoutManager
    }

    fun androidx.recyclerview.widget.RecyclerView.initVertical(adapter: androidx.recyclerview.widget.RecyclerView.Adapter<RecyclerViewHolder>) {
        initVertical()
        this.adapter = adapter
    }


    fun androidx.recyclerview.widget.RecyclerView.initGrid(count: Int): androidx.recyclerview.widget.GridLayoutManager {
        mContext = this.context
        this.layoutManager = RecycleLayout.RecycleViewBuilder.getGridLayoutManager(count, this)
        return this.layoutManager as androidx.recyclerview.widget.GridLayoutManager
    }

    fun androidx.recyclerview.widget.RecyclerView.initGrid(count: Int, adapter: androidx.recyclerview.widget.RecyclerView.Adapter<RecyclerViewHolder>) {
        initGrid(count)
        this.adapter = adapter
    }


    private object RecycleViewBuilder {
        /***
         * orientation 0横向 1竖向
         */
        fun getLinLayoutManager(orientation: Int, view: androidx.recyclerview.widget.RecyclerView): androidx.recyclerview.widget.LinearLayoutManager {
            val layoutManager = object : androidx.recyclerview.widget.LinearLayoutManager(mContext) {
                override fun onLayoutChildren(recycler: androidx.recyclerview.widget.RecyclerView.Recycler?, state: androidx.recyclerview.widget.RecyclerView.State?) {
                    super.onLayoutChildren(recycler, state)
                }
            }
            view.isNestedScrollingEnabled = false
            layoutManager.orientation = orientation
            return layoutManager
        }

        fun getGridLayoutManager(count: Int, view: androidx.recyclerview.widget.RecyclerView): androidx.recyclerview.widget.GridLayoutManager {
            view.isNestedScrollingEnabled = false
            return androidx.recyclerview.widget.GridLayoutManager(mContext, count)
        }
    }


}